deeplabcut/pose_estimation_3d/camera_calibration.py

```diff
@@ -105,35 +106,82 @@ def calibrate_cameras(config,cbrow = 8,cbcol = 6,calibrate=False,alpha=0.4):
+        msg = "No calibration images found. Make sure the calibration images are saved as .jpg and with prefix as the camera name as specified in the config.yaml file."
+        raise Exception(msg)
+
+    # Find incomplete pairs
+    corner_selected_by_user_check = glob.glob(os.path.join(path_corners, '*.jpg')) != [] and len(glob.glob(os.path.join(path_corners, '*.jpg'))) <= len(images)
+    if calibrate is True and corner_selected_by_user_check:
+        valid_corners = [Path(img_path).stem for img_path in glob.glob(os.path.join(path_corners, '*.jpg'))]
+
+        usable_corners = []
+        occurance = []
+        finder = re.compile(r'\d+')
+        img_id_warn = False
+        for img_name in valid_corners:
+            if cam_names[0] in img_name:
+                img_id_search = finder.findall(img_name[len(cam_names[0])+1:])
+            elif cam_names[1] in img_name:
+                img_id_search = finder.findall(img_name[len(cam_names[1])+1:])
+
+            if img_id_search != []:
+                if len(img_id_search) == 1:
+                    img_id = int(img_id_search[0])
+                else:
+                    if img_id_warn is False:
+                        img_id_warn = True
+                        warn('Detected multiple numbers in ID: %s\nUsing the last integer' % img_id_search)
+                    img_id = int(img_id_search[-1])
+            else:
+                warn('Could not detect ID of %s. Skipping' % img_name)
+                continue
+
+            if img_id in occurance:
+                usable_corners.append(img_id)
+            else:
+                occurance.append(img_id)
     
     for fname in images:
         for cam in cam_names:
             if cam in fname:
-                filename = Path(fname).stem
+                fname_pathlib = Path(fname)
+                img_name = fname_pathlib.stem
+
+                if calibrate is True and corner_selected_by_user_check:
+                    if cam_names[0] in img_name:
+                        img_id_search = finder.findall(img_name[len(cam_names[0])+1:])
+                    elif cam_names[1] in img_name:
+                        img_id_search = finder.findall(img_name[len(cam_names[1])+1:])
+                    img_id = int(img_id_search[-1])
+                    if img_id not in usable_corners:
+                        print('Skipping %s as it is part of a bad pair' % fname_pathlib.name)
+                        continue
+
                 img = cv2.imread(fname)
                 gray = cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
 
                 # Find the chess board corners
                 ret, corners = cv2.findChessboardCorners(gray, (cbcol,cbrow),None,) #  (8,6) pattern (dimensions = common points of black squares)
                 # If found, add object points, image points (after refining them)
-                if ret == True:
+                if ret is True:
```

Comment:
Because it is a Boolean test

I would have implemented this to every Boolean test in the package, but I followed your instruction to keep this PR strictly about the 3D subpackage, which I understand (I would do the same as you 👍 ).